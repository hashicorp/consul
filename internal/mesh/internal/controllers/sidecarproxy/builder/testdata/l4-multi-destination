{
  "proxyState": {
    "clusters": {
      "api-1.default.dc1.internal.foo.consul": {
        "endpointGroup": {
          "dynamic": {
            "config": {
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-1.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
                  ]
                }
              }
            }
          }
        }
      },
      "api-2.default.dc1.internal.foo.consul": {
        "endpointGroup": {
          "dynamic": {
            "config": {
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-2.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api2-identity"
                  ]
                }
              }
            }
          }
        }
      }
    },
    "identity": {
      "name": "test-identity",
      "tenancy": {
        "namespace": "default",
        "partition": "default",
        "peerName": "local"
      }
    },
    "listeners": [
      {
        "direction": "DIRECTION_OUTBOUND",
        "hostPort": {
          "host": "1.1.1.1",
          "port": 1234
        },
        "name": "api-1:tcp:1.1.1.1:1234",
        "routers": [
          {
            "l4": {
              "name": "api-1.default.dc1.internal.foo.consul",
              "statPrefix": "upstream.api-1.default.default.dc1"
            }
          }
        ]
      },
      {
        "direction": "DIRECTION_OUTBOUND",
        "name": "api-2:tcp:/path/to/socket",
        "routers": [
          {
            "l4": {
              "name": "api-2.default.dc1.internal.foo.consul",
              "statPrefix": "upstream.api-2.default.default.dc1"
            }
          }
        ],
        "unixSocket": {
          "mode": "0666",
          "path": "/path/to/socket"
        }
      }
    ]
  },
  "requiredEndpoints": {
    "api-1.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-1",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v1alpha1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    },
    "api-2.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-2",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v1alpha1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    }
  }
}