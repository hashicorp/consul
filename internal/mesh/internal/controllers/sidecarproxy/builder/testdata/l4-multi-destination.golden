{
  "proxyState":  {
    "identity":  {
      "tenancy":  {
        "partition":  "default",
        "namespace":  "default",
        "peerName":  "local"
      },
      "name":  "test-identity"
    },
    "listeners":  [
      {
        "name":  "api-1:tcp:1.1.1.1:1234",
        "direction":  "DIRECTION_OUTBOUND",
        "hostPort":  {
          "host":  "1.1.1.1",
          "port":  1234
        },
        "routers":  [
          {
            "l4":  {
              "name":  "api-1.default.dc1.internal.foo.consul",
              "statPrefix":  "upstream.api-1.default.default.dc1"
            }
          }
        ]
      },
      {
        "name":  "api-2:tcp:/path/to/socket",
        "direction":  "DIRECTION_OUTBOUND",
        "unixSocket":  {
          "path":  "/path/to/socket",
          "mode":  "0666"
        },
        "routers":  [
          {
            "l4":  {
              "name":  "api-2.default.dc1.internal.foo.consul",
              "statPrefix":  "upstream.api-2.default.default.dc1"
            }
          }
        ]
      }
    ],
    "clusters":  {
      "api-1.default.dc1.internal.foo.consul":  {
        "endpointGroup":  {
          "dynamic":  {
            "config":  {
              "disablePanicThreshold":  true
            },
            "outboundTls":  {
              "outboundMesh":  {
                "identityKey":  "test-identity",
                "validationContext":  {
                  "spiffeIds":  [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
                  ]
                },
                "sni":  "api-1.default.dc1.internal.foo.consul"
              }
            }
          }
        }
      },
      "api-2.default.dc1.internal.foo.consul":  {
        "endpointGroup":  {
          "dynamic":  {
            "config":  {
              "disablePanicThreshold":  true
            },
            "outboundTls":  {
              "outboundMesh":  {
                "identityKey":  "test-identity",
                "validationContext":  {
                  "spiffeIds":  [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api2-identity"
                  ]
                },
                "sni":  "api-2.default.dc1.internal.foo.consul"
              }
            }
          }
        }
      }
    }
  },
  "requiredEndpoints":  {
    "api-1.default.dc1.internal.foo.consul":  {
      "id":  {
        "name":  "api-1",
        "type":  {
          "group":  "catalog",
          "groupVersion":  "v1alpha1",
          "kind":  "ServiceEndpoints"
        },
        "tenancy":  {
          "partition":  "default",
          "namespace":  "default",
          "peerName":  "local"
        }
      },
      "port":  "tcp"
    },
    "api-2.default.dc1.internal.foo.consul":  {
      "id":  {
        "name":  "api-2",
        "type":  {
          "group":  "catalog",
          "groupVersion":  "v1alpha1",
          "kind":  "ServiceEndpoints"
        },
        "tenancy":  {
          "partition":  "default",
          "namespace":  "default",
          "peerName":  "local"
        }
      },
      "port":  "tcp"
    }
  }
}